### ISSUE TITLE
[Wildcard Variables] implement `THIS_ACCESS_FROM_FIELD_INITIALIZER`

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#56442]
The front-end has this error:

```
ThisAccessInFieldInitializer:
  problemMessage: "Can't access 'this' in a field initializer to read '#name'."
  analyzerCode: THIS_ACCESS_FROM_FIELD_INITIALIZER
```

that names `THIS_ACCESS_FROM_FIELD_INITIALIZER` (which doesn't exist).

That said, we *do* have [`IMPLICIT_THIS_REFERENCE_IN_INITIALIZER`](https://dart.dev/tools/diagnostic-messages?utm_source=dartdev&utm_medium=redir&utm_id=diagcode&utm_content=implicit_this_reference_in_initializer#implicit_this_reference_in_initializer).

@kallentu: any reason we need a new error code here and can't use `IMPLICIT_THIS_REFERENCE_IN_INITIALIZER`?

Here's the relevant part of the spec:

https://github.com/dart-lang/language/blob/main/working/wildcards/feature-specification.md#initializing-formals

A corresponding (failing) test:

https://dart-ci.appspot.com/log/any/analyzer-asserts-linux/25960/language/wildcard_variables/initializing_formal_parameters/this_initializer_access_error_test

(There are co19 ones too.)

**Comment by pq** [comment]
Ok, after a nice ling chat w/ @stereotype441, I think I see what's happening.

**TL;DR** `THIS_ACCESS_FROM_FIELD_INITIALIZER` is likely a stale reference to an analyzer error code that existing pre CFE. 

I'll loop back and do some cleanup but for now this whole issue is a bit of a red herring so I'll close it out.

Sorry for the noise!
